src/cdk/datetime/datepicker-input.ts

```diff
@@ -0,0 +1,345 @@
+    '(change)': '_onChange()',
+    '(blur)': '_onBlur()',
+  },
+  exportAs: 'cdkDatepickerInput',
+})
+export class CdkDatepickerInput<D> implements AfterContentInit, ControlValueAccessor, OnDestroy,
+  Validator {
+  /** The datepicker that this input is associated with. */
+  @Input('cdkDatepicker')
+  set datepicker(value: CdkDatepicker<D>) {
+    this._registerDatepicker(value);
+  }
+  _datepicker: CdkDatepicker<D>;
+
+  /** Registering datepicker with this input. */
+  private _registerDatepicker(value: CdkDatepicker<D>) {
+    if (value) {
+      this._datepicker = value;
+      this._datepicker._registerInput(this);
+    }
+  }
+
+  /** Function that can be used to filter out dates within the datepicker. */
+  @Input()
+  set cdkDatepickerFilter(value: (date: D | null) => boolean) {
+    this._dateFilter = value;
+    this._validatorOnChange();
+  }
+  _dateFilter: (date: D | null) => boolean;
+
+  /** The value of the input. */
+  @Input()
+  get value(): D | null { return this._value; }
+  set value(value: D | null) {
+    value = this._dateAdapter.deserialize(value);
+    this._lastValueValid = !value || this._dateAdapter.isValid(value);
+    value = this._getValidDateOrNull(value);
+    const oldDate = this.value;
+    this._value = value;
+    this._formatValue(value);
+
+    if (!this._dateAdapter.sameDate(oldDate, value)) {
+      this._valueChange.emit(value);
+    }
+  }
+  private _value: D | null;
+
+  /** The minimum valid date. */
+  @Input()
+  get min(): D | null { return this._min; }
+  set min(value: D | null) {
+    this._min = this._getValidDateOrNull(this._dateAdapter.deserialize(value));
+    this._validatorOnChange();
+  }
+  private _min: D | null;
+
+  /** The maximum valid date. */
+  @Input()
+  get max(): D | null { return this._max; }
+  set max(value: D | null) {
+    this._max = this._getValidDateOrNull(this._dateAdapter.deserialize(value));
+    this._validatorOnChange();
+  }
+  private _max: D | null;
+
+  /** Whether the datepicker-input is disabled. */
+  @Input()
+  get disabled(): boolean { return !!this._disabled; }
+  set disabled(value: boolean) {
+    const newValue = coerceBooleanProperty(value);
+    const element = this._elementRef.nativeElement;
+
+    if (this._disabled !== newValue) {
+      this._disabled = newValue;
+      this._disabledChange.emit(newValue);
+    }
+
+    // We need to null check the `blur` method, because it's undefined during SSR.
+    if (newValue && element.blur) {
+      // Normally, native input elements automatically blur if they turn disabled. This behavior
+      // is problematic, because it would mean that it triggers another change detection cycle,
+      // which then causes a changed after checked error if the input element was focused before.
+      element.blur();
+    }
+  }
+  private _disabled: boolean;
+
+  /** Emits when a `change` event is fired on this `<input>`. */
+  @Output() readonly dateChange: EventEmitter<DatepickerInputEvent<D>> =
```

Comment:
The type isn't necessary here since it's inferred from the assignment

(for this and other members)